tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,157 @@
+					})
+
+					Specify("the istio resource version match the applied version", func() {
+						// TODO: we need to add a function to get the istio version from the control panel directly
+						// and compare it with the applied version
+						// This is a TODO because actual version.yaml contains for example latest and not the actual version
+						// Posible solution is to add actual version to the version.yaml
+					})
+
+					Specify("istio resource stopped reconciling", func() {
+						istiodPodName, _ := kubectl.GetPodFromLabel(controlPlaneNamespace, "app=istiod")
+						Eventually(kubectl.GetPodLogs).WithArguments(controlPlaneNamespace, istiodPodName, "30s").ShouldNot(ContainSubstring("Reconciliation done"))
+						GinkgoWriter.Println("Istio Operator stopped reconciling")
+					})
+
+					It("deploys istiod", func() {
+						expectedDeployments := []string{"istiod"}
+
+						deploymentsList, err := kubectl.GetDeployments(controlPlaneNamespace)
+						if err != nil {
+							Fail("Error getting deployments")
+						}
+
+						Expect(deploymentsList).To(Equal(expectedDeployments), "Deployments List should contains only istiod")
+					})
+
+					It("deploys the CNI DaemonSet when running on OpenShift", func() {
+						if ocp == "true" {
+							Eventually(kubectl.GetDaemonSets).WithArguments(namespace).Should(ContainElement("istio-cni-node"), "CNI DaemonSet should be deployed")
+							Eventually(kubectl.GetPodPhase).WithArguments(namespace, "k8s-app=istio-cni-node").Should(Equal("Running"), "CNI DaemonSet should be Running")
+							GinkgoWriter.Println("CNI DaemonSet is deployed in the namespace and Running")
+						} else {
+							Eventually(kubectl.GetDaemonSets).WithArguments(namespace).Should(BeEmpty(), "CNI DaemonSet should not be deployed on OpenShift")
+							GinkgoWriter.Println("CNI DaemonSet is not deployed in the namespace because it's not OpenShift")
+						}
+					})
+				})
+
+				When("the Istio CR is deleted", func() {
+					BeforeEach(func() {
+						Eventually(deleteIstioCR).WithArguments(version).Should(Succeed(), "Istio CR should be deleted")
+						GinkgoWriter.Println("Istio CR's deleted")
+					})
+
+					Specify("the namespace is empty", func() {
+						Eventually(kubectl.GetAllResources).WithArguments(controlPlaneNamespace).Should(Equal(kubectl.EmptyResourceList), "Namespace should be empty")
+					})
+				})
+			})
+		}
+	})
+
+	Describe("operator uninstallation", func() {
+		When("is deleted the operator manifest from the cluster", func() {
+			BeforeEach(func() {
+				Eventually(undeployOperator).Should(Succeed(), "Operator deployment should be deleted")
+				GinkgoWriter.Println("Operator deployment is deleted")
+			})
+
+			Specify("the operator it's deleted", func() {
+				Eventually(kubectl.GetPodFromLabel).WithArguments(namespace, "control-plane=istio-operator").Should(BeEmpty(), "Istio-operator pod should be deleted")
+				GinkgoWriter.Println("Istio-operator pod is deleted")
+			})
+		})
```

Comment:
Do we even need this `When`? We're not testing the operator here; we're just performing clean-up. This could be replaced with a simple `AfterAll` and we probably don't need to check that the namespace is empty, since it's Kubernetes/OpenShift that's performing the deletion.

Reply:
We can't move this to an `AfterAll` because the previous Spec inside the Describe is doing the iteration over all the versions, and the previous Spec that is the Operator Installation is outside of that iteration also. We need that different Spec to be able to properly do the iteration with issues